networkx/algorithms/operators/unary.py

```diff
@@ -67,3 +68,46 @@ def reverse(G, copy=True):
         raise nx.NetworkXError("Cannot reverse an undirected graph.")
     else:
         return G.reverse(copy=copy)
+
+
+@not_implemented_for('directed')
+@not_implemented_for('multigraph')
+def mycielski(G, iterations=1):
+    """Returns the Mycielskian of a simple, undirected graph G
+
+    The Mycielskian of graph preserves a graph's triangle free
+    property while increasing the chromatic number by 1.
+
+    Parameters
+    ----------
+    G : graph
+        A simple, undirected NetworkX graph
+    iterations : int
+        The number of iterations of the Mycielski operation to
+        preform on G. Defaults to 1. Must be a non-negative integer.
+
+    Returns
+    -------
+    M : graph
+        The Mycielskian of G after the specified number of iterations.
+
+    Notes
+    ------
+    Graph, node, and edge data are not necessarily propagated to the new graph.
+
+    """
+    if not isinstance(iterations, int) or iterations < 1:
+        raise nx.NetworkXError('iterations must be a non-negative integer')
+    
+    n = G.number_of_nodes()
+    M = nx.convert_node_labels_to_integers(G)
+    M.add_nodes_from(range(n, 2*n))
+    old_edges = list(M.edges())
+    M.add_edges_from((e[0], e[1]+n) for e in old_edges)
+    M.add_edges_from((e[0]+n, e[1]) for e in old_edges)
+    M.add_node(2*n)
+    M.add_edges_from((u+n, 2*n) for u in range(n))
+
+    for i in range(iterations-1):
+        M=mycielski(M)
```

Comment:
Please use spaces around the assignment operator. Run `flake8` on your code to fix style errors.